File: 2022-09-20/convoglio/att/template.py
#!/usr/bin/env python3
import sys
sys.setrecursionlimit(10**9)

MAXN=100000
MAXM=200000
N, M = map(int,input().strip().split())
assert 0 <= N <= MAXN
assert 0 <= M <= MAXM
#print(f"{N=}, {M=}")
for i in range(M):
    u,v = map(int,input().strip().split())
    assert 0 <= u < N
    assert 0 <= v < N
    #print(f"message={u}, ship={v}")
    
print(-1)
exit(0)

File: 2022-09-20/convoglio/att/template.cpp
#include <cassert>
#include <cstdio>
using namespace std;

const int MAXM=200000, MAXN=100000;
int N,M;

int main(){
	scanf("%d%d",&N,&M);
	assert(M >= 2);
	assert(M <= MAXM);
	assert(N >= 2);
	assert(N < MAXN);
	//printf("N=%d, M=%d\n",N,M);
	for(int i=0;i<M;i++){
		int a,b;
		scanf("%d%d",&a,&b);
		assert(a >= 0);
		assert(a < N);
		assert(b >= 0);
		assert(b < N);
		//printf("message=%d, ship=%d\n",a,b);
	}
	printf("-1\n");
	return 0;
}
